Portfolio: setup and mobile version skeleton

OmarRamoun opened the pull request:
## Enhancement Needed :recycle: 

- :heavy_plus_sign: Add Structure & Style of Header Section with Logo and Humbergur Menu :hamburger: .
- :heavy_plus_sign: Add Structure & Style of Intro Section With info About the developer :information_source: .
- :heavy_plus_sign: :link: Add  Structure & Style of  Social Media Links :busts_in_silhouette: .
- :heavy_plus_sign: Add Structure & Style of Project Assets :framed_picture: .

Commit:
Update package-lock.json and package.json files with new veresions of packages

Commit:
Add assets folder

folder containing all fonts used in the project

Commit:
Add images folder and favicon icon

images folder containing all images used in the project

Commit:
Update screenshot image of the app

Commit:
Update the readme file

Add project information to the readme file

Commit:
Add page description to the head

Commit:
Add stylesheets links to the head element

Commit:
Update document title

Commit:
Add favicon to the document

Commit:
Merge branch 'feature/add-head-element-information' into develop

Commit:
Add wrapper to the header section

Commit:
Add logo to the header section

Commit:
Add navbar hamburger icon to the header section

Commit:
Add surrounding comments around header section

to seprate header section logically from other parts of the page

Commit:
Merge branch 'feature/add-page-header' into develop

Commit:
Resize logo image

for a more optimized solution

Commit:
Add wrapper to intro section

Commit:
Add intro section's introductory heading and paragraph

a small introductory paragraph about the owner of the portfolio

Commit:
Add link to contact me form

Commit:
Add social media links

Commit:
Add surrounding comments around intro section

to seprate intro section logically from other parts of the page

Commit:
Merge branch 'feature/add-page-intro-section' into develop

Commit:
Add CSS table of contents

to help track down code easily

Commit:
Add root element styles

Commit:
Add body element styles

Commit:
Add .bg class styles

Commit:
Add white color to root element's variables

Commit:
Add bg class to header and intro section

Commit:
Add logo styling

Commit:
Add .hero class styles

hero class is a common class that is used to style intro section and about section

Commit:
Add intro section styling

Commit:
Add hero class to intro section

Commit:
Merge branch 'feature/add-page-style-css' into develop

Commit:
Add CSS table of contents

to help track down code easily

Commit:
Add general rules to layout page elements

Commit:
Add body layout rules

Commit:
Add header section layout

Commit:
Add wrapper layout

Commit:
Add intro section

Commit:
Add social icons layout to the gernal rules part

Commit:
Adjust intro section top padding

to narrow down the space between the logo and the introductory text

Commit:
Merge branch 'feature/add-page-layout-css' into develop

Commit:
Add space around comments

to follow stylelint guidelines

Commit:
Add space around comments

to follow stylelint guidelines

Commit:
Merge branch 'bugfix/fix-space-around-comments' into develop

barackm commented on index.html:
```diff
@@ -6,15 +6,47 @@
+        <a href="https://medium.com/@omartarekramoun"><img src="./images/social/medium.svg" alt=""></a>
+        <a href="https://github.com/mrramoun"><img src="./images/social/github.svg" alt=""></a>
+        <a href="https://angel.co/u/omar-ramoun"><img src="./images/social/vector.svg" alt=""></a>
+      </div>
```
- [x] This is great to add social media icons to your project, However, it could be nicer to wrap all the icons in an unordered list `<ul>` because this is supposed to be a list of icons, see the example below: 👇 
```
  <ul>
    <li>
        <a href="https://twitter.com/ramoun16"><img src="./images/social/twitter.svg" alt=""></a>
    </li>
    <li>
        <a href="https://github.com/mrramoun"><img src="./images/social/github.svg" alt=""></a>
    </li>
  </ul>
```
I promise this is going to make your code more clean and sematic 👌  😉

barackm commented on css/style.css:
```diff
@@ -1,19 +1,75 @@
+
+/* about/start */
+
+/* about/end */
```
- [x] It is a good fing to add comments for reference, but for now, we can see that these comments are not necessary because they try to introduce something that does not exist yet. Please let's remove unnecessary comments to make our code clean.

barackm commented on css/layout.css:
```diff
@@ -1,3 +1,81 @@
+  flex-wrap: wrap;
+  justify-content: space-between;
+  padding: 0 1.3rem;
+  width: 100%;
```
- [x] [OPTIONAL] This is so fantastic to use CSS flexbox to layout your content, but there is just one more thing you can do to improve your code, that is DRY(don't repeat yourself). I can see that you have used the property `display: flex` at multiple places, please let's use DRY to keep the code clean by extracting the property into a separate class that can be reused at multiple places almost like what you did for the variables 👏 , see the example below: 

- CSS file
```
.d-flex {
  display: flex;
}

.flex-center {
  align-items: center;
  justify-content: center;
}
```
- HTML file
```
<div class="d-flex flex-center">{your content}</div>
```

OmarRamoun replied:
I like this one. That's an awesome tip. Really, thank you.<3

barackm commented on index.html:
```diff
@@ -6,15 +6,47 @@
+      <div id="logo">
+        <img src="./images/darwin.jpg" alt="">
+        <a href="./">Ramoun</a>
+      </div>
```
- [x] [OPTIONAL] This was a very good idea to have the logo as an anchor tag `<a>`, I think this could be even better if you could just take also the image inside the `<a>` tag because people may take the image as part of the logo and try to click on it. you may have something like this instead 
```
<a href="#" class="logo">
    <img src="./images/darwin.jpg" alt="">
    <span>Ramoun</span>
</a>
```
With this, by clicking on the image we should have the same behavior as clicking on the name. 😉

barackm reviewed (changes requested):
Hi @MrRamoun , 👋 

You have done an amazing job so far! 👏👏👏
However, there are some issues that you still need to work on to go to the next project. But remember you are almost there 😉  😉 !

### To highlight

✔️  Very descriptive PR title
✔️  Very descriptive PR description
✔️  Super descriptive README
✔️  Super descriptive commit messages 
✔️  Use of semantic HTML

You have just done a great job 👏  👏  despite some small changes we need make

# STATUS: Required Changes ♻️
### _Check the comments under the review._ 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Commit:
Replace div container for social bar with a ul element

and update the links with list items

Commit:
Add a rule to remove list items automatic styles

Commit:
Remove useless comments

Commit:
Add new flex classes to the html elements

to make the code more DRY(don't repeat your self). Plus replacing the logo id with a class in order to be able to add mulitple classes to it

Commit:
Move the hero up to the general rules section

Commit:
Replacing the logo id with the logo class

in order to add more classes to the logo selector

Commit:
Remvoe repeated font weight property

and move it to a new class '.bold' to make the code more DRY!

Commit:
Move repeated code from the hero children rule-sets into a seprate class

to make the code more DRY!

Commit:
Make the image clickable

since the image is part of the logo

Commit:
Merge branch 'feature/improve-code-semantics' into develop

DanteAlonsoHT commented on README.md:
```diff
@@ -1,27 +1,25 @@
-> One paragraph statement about the project.
+> A Simple Portfolio App for a Web developer.
 
 ![screenshot](images/app_screenshot.png)
```
I know you did a good effort doing your description for this project, but I know you can do a more impressive README file, if you want to have a reference, 3 lines or more would be nice, just try to think about what you did, what kind of technologies or techniques you used, what is the scope for this project, and so on.

- [x] Improve your description for your project.

DanteAlonsoHT reviewed (changes requested):
<h1>Status: Changes Required 🚫 ♻️  </h1>

Hi @MrRamoun, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there is one mandatory change required, I know you have strong capabilities to finish this change required in time 😉 

 ## Required Change ♻️

_Check the comment under the review._

## Optional suggestions

**Nothing, you did an amazing job!**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

Commit:
Add more information to the README file

to make it more descriptive and professional

AshakaE reviewed (approved):
### Hi @MrRamoun  🙋‍♀️, 

# STATUS APPROVED ✔
Thanks for making the required changes
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
